[Java15] Test Failure (20200914-0243): com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkFfdc

RTC 277839

```
junit.framework.AssertionFailedError: 2020-09-14-02:04:46:598 Test failed with one or more errors
at com.ibm.ws.logging.json.fat.JSONEventsTest.checkFfdc(JSONEventsTest.java:118)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

```
[09/11/2020 05:31:46:986 PDT] 001 LibertyServer                  startServerWithArgs            I >>> STARTING SERVER: com.ibm.ws.logging.json.MessagesLogEnvServer
[09/11/2020 05:31:46:986 PDT] 001 LibertyServer                  startServerWithArgs            I Starting com.ibm.ws.logging.json.MessagesLogEnvServer; clean=true, validateApps=true, expectStartFailure=false, cmd=start, args=null
[09/11/2020 05:31:46:986 PDT] 001 LibertyServer                  startServerWithArgs            I Tidying logs
[09/11/2020 05:31:46:988 PDT] 001 LibertyServer                  calculateDebugInfo             I debugging allowed for server = com.ibm.ws.logging.json.MessagesLogEnvServer, but didn't find valid port, so debug not enabled
[09/11/2020 05:31:46:989 PDT] 001 LibertyServer                  startServerWithArgs            I Using additional env props: {JVM_ARGS= -Djava.io.tmpdir=/home/jazz_build/Build/jbe/build/dev/build.image/build/tmp -Djava.security.egd=file:///dev/urandom, JAVA_HOME=/home/jazz_build/Build/jbe/jvms/openjdk15_hotspot_ea_b32-sdk, LOG_FILE=MessagesLogEnvTest.log, LOG_DIR=/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/}
[09/11/2020 05:31:51:203 PDT] 001 LibertyServer                  validateServerStarted          I Waiting up to 30 seconds for server confirmation:  CWWKF0011I to be found in /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/MessagesLogEnvTest.log
[09/11/2020 05:31:51:203 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/MessagesLogEnvTest.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/MessagesLogEnvTest.log
[09/11/2020 05:31:51:239 PDT] 001 LibertyServer                  validateAppsLoaded             I Started searching for app manager messages at 5:31:51 AM PDT and finished at 5:31:51 AM PDT
[09/11/2020 05:31:51:260 PDT] 001 FeatureDependencyProcessor     validateTestedFeatures         I Validated that all installed features were present in test dependencies JSON file.
[09/11/2020 05:31:51:261 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log
[09/11/2020 05:31:51:264 PDT] 001 FATRunner                      evaluate                       I entering com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkMessage
[09/11/2020 05:31:51:265 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log
[09/11/2020 05:31:51:272 PDT] 001 FATRunner                      evaluate                       I exiting com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkMessage
[09/11/2020 05:31:51:273 PDT] 001 FATRunner                      evaluate                       I entering com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkAccessLog
[09/11/2020 05:31:51:273 PDT] 001 LibertyServer                  addInstalledAppForValidation   I Adding installed app: LogstashApp for validation
[09/11/2020 05:31:51:577 PDT] 001 LibertyServer                  waitForMultipleStringsInLog    I Started waiting for 1 messages matching regexp [ CWWKZ0001I:.*LogstashApp] at 5:31:51 AM PDT and finished at 5:31:51 AM PDT
[09/11/2020 05:31:51:577 PDT] 001 LibertyServer                  searchForMessages              I installApp message appears in log 1 time(s)
[09/11/2020 05:31:51:578 PDT] 001 TestUtils                      runApp                         I ---> Running the application with url : http://localhost:8010/LogstashApp/AccessURL
[09/11/2020 05:31:51:578 PDT] 001 TestUtils                      runGetMethod                   I URL = http://localhost:8010/LogstashApp/AccessURL
[09/11/2020 05:31:51:715 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log
[09/11/2020 05:31:51:734 PDT] 001 FATRunner                      evaluate                       I exiting com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkAccessLog
[09/11/2020 05:31:51:735 PDT] 001 FATRunner                      evaluate                       I entering com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkFfdc
[09/11/2020 05:31:51:736 PDT] 001 LibertyServer                  addInstalledAppForValidation   I Adding installed app: LogstashApp for validation
[09/11/2020 05:32:22:300 PDT] 001 LibertyServer                  waitForMultipleStringsInLog    I Started waiting for 2 messages matching regexp [ CWWKZ0001I:.*LogstashApp] at 5:31:51 AM PDT and finished at 5:32:22 AM PDT
[09/11/2020 05:32:22:301 PDT] 001 LibertyServer                  searchForMessages              I Incorrect number of installApp messages in the log.  An error may have occurred.
[09/11/2020 05:32:22:301 PDT] 001 TestUtils                      runApp                         I ---> Running the application with url : http://localhost:8010/LogstashApp?isFFDC=true
[09/11/2020 05:32:22:301 PDT] 001 TestUtils                      runGetMethod                   I URL = http://localhost:8010/LogstashApp?isFFDC=true
[09/11/2020 05:32:22:499 PDT] 001 TestUtils                      runApp                         I  ---> Exception : Server returned HTTP response code: 500 for URL: http://localhost:8010/LogstashApp?isFFDC=true
[09/11/2020 05:32:22:499 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log
[09/11/2020 05:32:22:505 PDT] 001 JSONEventsTest                 checkJsonMessage               I Invalid keys found:[message]
[09/11/2020 05:32:22:505 PDT] 001 JSONEventsTest                 checkFfdc                      I Message line:{"type":"liberty_ffdc","host":"ebcprh2551ey4vagm-n.fyre.ibm.com","ibm_userDir":"\/home\/jazz_build\/Build\/jbe\/build\/dev\/image\/output\/wlp\/usr\/","ibm_serverName":"com.ibm.ws.logging.json.MessagesLogEnvServer","ibm_datetime":"2020-09-11T05:32:22.337-0700","message":"Cannot invoke \"String.toString()\" because \"myString\" is null","ibm_className":"com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters","ibm_exceptionName":"java.lang.NullPointerException","ibm_probeID":"1105","ibm_threadId":"0000001f","ibm_stackTrace":"java.lang.NullPointerException: Cannot invoke \"String.toString()\" because \"myString\" is null\n\tat com.ibm.logs.LogstashServlet.doGet(LogstashServlet.java:57)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:575)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:668)\n\tat com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)\n\tat com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:729)\n\tat com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:426)\n\tat com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1226)\n\tat com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5057)\n\tat com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)\n\tat com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1005)\n\tat com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)\n\tat com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1134)\n\tat com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:415)\n\tat com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:374)\n\tat com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:577)\n\tat com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:511)\n\tat com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:359)\n\tat com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:326)\n\tat com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)\n\tat com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)\n\tat com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)\n\tat com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)\n\tat com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)\n\tat com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)\n\tat com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)\n\tat java.base\/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\n\tat java.base\/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\n\tat java.base\/java.lang.Thread.run(Thread.java:832)\n","ibm_objectDetails":"Object type = com.ibm.ws.webcontainer.osgi.filter.WebAppFilterManagerImpl\n  _filterWrappers = class java.util.Hashtable@1b6461c2\n    table = \"\/* Could not access table *\/\"\n    count = \"\/* Could not access count *\/\"\n    threshold = \"\/* Could not access threshold *\/\"\n    loadFactor = \"\/* Could not access loadFactor *\/\"\n    modCount = \"\/* Could not access modCount *\/\"\n    serialVersionUID = \"\/* Could not access serialVersionUID *\/\"\n    MAX_ARRAY_SIZE = \"\/* Could not access MAX_ARRAY_SIZE *\/\"\n    keySet = \"\/* Could not access keySet *\/\"\n    entrySet = \"\/* Could not access entrySet *\/\"\n    values = \"\/* Could not access values *\/\"\n    KEYS = \"\/* Could not access KEYS *\/\"\n    VALUES = \"\/* Could not access VALUES *\/\"\n    ENTRIES = \"\/* Could not access ENTRIES *\/\"\n  chainCache = class java.util.Collections$SynchronizedMap@6b702f0b\n    serialVersionUID = \"\/* Could not access serialVersionUID *\/\"\n    m = \"\/* Could not access m *\/\"\n    mutex = \"\/* Could not access mutex *\/\"\n    keySet = \"\/* Could not access keySet *\/\"\n    entrySet = \"\/* Could not access entrySet *\/\"\n    values = \"\/* Could not access values *\/\"\n  forwardChainCache = class java.util.Collections$SynchronizedMap@15a558e9\n    serialVersionUID = \"\/* Could not access serialVersionUID *\/\"\n    m = \"\/* Could not access m *\/\"\n    mutex = \"\/* Could not access mutex *\/\"\n    keySet = \"\/* Could not access keySet *\/\"\n    entrySet = \"\/* Could not access entrySet *\/\"\n    values = \"\/* Could not access values *\/\"\n  includeChainCache = class java.util.Collections$SynchronizedMap@3c52edd4\n    serialVersionUID = \"\/* Could not access serialVersionUID *\/\"\n    m = \"\/* Could not access m *\/\"\n    mutex = \"\/* Could not access mutex *\/\"\n    keySet = \"\/* Could not access keySet *\/\"\n    entrySet = \"\/* Could not access entrySet *\/\"\n    values = \"\/* Could not access values *\/\"\n  errorChainCache = class java.util.Collections$SynchronizedMap@480e1ed\n    serialVersionUID = \"\/* Could not access serialVersionUID *\/\"\n    m = \"\/* Could not access m *\/\"\n    mutex = \"\/* Could not access mutex *\/\"\n...","ibm_sequence":"1599827542337_0000000000001"}
[09/11/2020 05:32:22:506 PDT] 001 FATRunner                      evaluate                       S null
java.lang.AssertionError: Test failed with one or more errors
	at org.junit.Assert.fail(Assert.java:93)
	at com.ibm.ws.logging.json.fat.JSONEventsTest.checkFfdc(JSONEventsTest.java:118)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:30)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:39)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:520)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1060)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:911)
[09/11/2020 05:32:22:509 PDT] 001 FATRunner                      evaluate                       I exiting com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkFfdc
[09/11/2020 05:32:22:518 PDT] 001 FATRunner                      evaluate                       I entering com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkTrace
[09/11/2020 05:32:22:526 PDT] 001 LibertyServer                  addInstalledAppForValidation   I Adding installed app: LogstashApp for validation
[09/11/2020 05:32:53:231 PDT] 001 LibertyServer                  waitForMultipleStringsInLog    I Started waiting for 3 messages matching regexp [ CWWKZ0001I:.*LogstashApp] at 5:32:22 AM PDT and finished at 5:32:53 AM PDT
[09/11/2020 05:32:53:231 PDT] 001 LibertyServer                  searchForMessages              I Incorrect number of installApp messages in the log.  An error may have occurred.
[09/11/2020 05:32:53:231 PDT] 001 TestUtils                      runApp                         I ---> Running the application with url : http://localhost:8010/LogstashApp/TraceURL
[09/11/2020 05:32:53:232 PDT] 001 TestUtils                      runGetMethod                   I URL = http://localhost:8010/LogstashApp/TraceURL
[09/11/2020 05:32:53:290 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log
[09/11/2020 05:32:53:298 PDT] 001 FATRunner                      evaluate                       I exiting com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkTrace
[09/11/2020 05:32:53:299 PDT] 001 FATRunner                      evaluate                       I entering com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkExtensions
[09/11/2020 05:32:53:299 PDT] 001 LibertyServer                  addInstalledAppForValidation   I Adding installed app: LogstashApp for validation
[09/11/2020 05:33:23:969 PDT] 001 LibertyServer                  waitForMultipleStringsInLog    I Started waiting for 4 messages matching regexp [ CWWKZ0001I:.*LogstashApp] at 5:32:53 AM PDT and finished at 5:33:23 AM PDT
[09/11/2020 05:33:23:969 PDT] 001 LibertyServer                  searchForMessages              I Incorrect number of installApp messages in the log.  An error may have occurred.
[09/11/2020 05:33:23:969 PDT] 001 TestUtils                      runApp                         I ---> Running the application with url : http://localhost:8010/LogstashApp/ExtURL
[09/11/2020 05:33:23:969 PDT] 001 TestUtils                      runGetMethod                   I URL = http://localhost:8010/LogstashApp/ExtURL
[09/11/2020 05:33:24:004 PDT] 001 LogMonitor                     getMarkOffset                  I Mark offset for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log=0 for /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.logging.json.MessagesLogEnvServer/logs/messages.log
[09/11/2020 05:33:24:014 PDT] 001 FATRunner                      evaluate                       I exiting com.ibm.ws.logging.json.fat.MessagesLogEnvTest.checkExtensions
[09/11/2020 05:33:24:016 PDT] 001 LibertyServer                  stopServer                     I <<< STOPPING SERVER: com.ibm.ws.logging.json.MessagesLogEnvServer
[09/11/2020 05:33:26:044 PDT] 001 LibertyServer                  stopServer                     I Stop Server Response: 
Stopping server com.ibm.ws.logging.json.MessagesLogEnvServer.
Server com.ibm.ws.logging.json.MessagesLogEnvServer stopped.
```

--- a/dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java
+++ b/dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java
@@ -101,7 +101,8 @@ public abstract class JSONEventsTest {
                                                                                       "ibm_sequence", "ibm_className", "ibm_exceptionName", "ibm_probeID",
                                                                                       "ibm_threadId", "ibm_stackTrace", "ibm_objectDetails"));
 
-        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>();
+        //JDK 15 has message as an additional field in ffdc events
+        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>(Arrays.asList("message"));
 
         getServer().addInstalledAppForValidation(APP_NAME);
         TestUtils.runApp(getServer(), "ffdc1");

--- a/dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java
+++ b/dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017 IBM Corporation and others.
+ * Copyright (c) 2017, 2020 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at

--- a/dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java
+++ b/dev/com.ibm.ws.logging.json_fat/fat/src/com/ibm/ws/logging/json/fat/JSONEventsTest.java
@@ -27,6 +27,8 @@ import com.ibm.websphere.simplicity.RemoteFile;
 import com.ibm.websphere.simplicity.log.Log;
 
 import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.MaximumJavaLevel;
+import componenttest.annotation.MinimumJavaLevel;
 import componenttest.topology.impl.LibertyServer;
 
 /**
@@ -94,6 +96,7 @@ public abstract class JSONEventsTest {
 
     @Test
     @ExpectedFFDC({ "java.lang.NullPointerException" })
+    @MaximumJavaLevel(javaLevel = 14)
     public void checkFfdc() throws Exception {
         final String method = "checkFfdc";
 
@@ -101,8 +104,37 @@ public abstract class JSONEventsTest {
                                                                                       "ibm_sequence", "ibm_className", "ibm_exceptionName", "ibm_probeID",
                                                                                       "ibm_threadId", "ibm_stackTrace", "ibm_objectDetails"));
 
+        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>();
+
+        getServer().addInstalledAppForValidation(APP_NAME);
+        TestUtils.runApp(getServer(), "ffdc1");
+
+        String line = getServer().waitForStringInLog("\\{.*\"type\":\"liberty_ffdc\".*\\}", getLogFile());
+        assertNotNull("Cannot find \"type\":\"liberty_ffdc\" from messages.log", line);
+
+        JsonReader reader = Json.createReader(new StringReader(line));
+        JsonObject jsonObj = reader.readObject();
+        reader.close();
+
+        if (!checkJsonMessage(jsonObj, ffdcKeysMandatoryList, ffdcKeysOptionalList)) {
+            Log.info(c, method, "Message line:" + line);
+            Assert.fail("Test failed with one or more errors");
+        }
+
+    }
+
+    @Test
+    @ExpectedFFDC({ "java.lang.NullPointerException" })
+    @MinimumJavaLevel(javaLevel = 15)
+    public void checkFfdcJava15() throws Exception {
+        final String method = "checkFfdc";
+
         //JDK 15 has message as an additional field in ffdc events
-        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>(Arrays.asList("message"));
+        ArrayList<String> ffdcKeysMandatoryList = new ArrayList<String>(Arrays.asList("ibm_datetime", "type", "host", "ibm_userDir", "ibm_serverName",
+                                                                                      "ibm_sequence", "ibm_className", "ibm_exceptionName", "ibm_probeID",
+                                                                                      "ibm_threadId", "ibm_stackTrace", "ibm_objectDetails", "message"));
+
+        ArrayList<String> ffdcKeysOptionalList = new ArrayList<String>();
 
         getServer().addInstalledAppForValidation(APP_NAME);
         TestUtils.runApp(getServer(), "ffdc1");

